velox/functions/lib/MysqlDateTime.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/lib/MysqlDateTime.h"
+#include <string>
+#include "velox/functions/lib/DateTimeFormatter.h"
+
+namespace facebook::velox::functions {
+MysqlDateTime::MysqlDateTime(const std::string& format) {
+  DateTimeFormatterBuilder builder(format.size());
+  if (format.empty()) {
+    VELOX_USER_FAIL("Invalid pattern specification.");
+  }
+
+  const char* cur = format.data();
+  const char* end = cur + format.size();
+  while (cur < end) {
+    auto tokenEnd = cur;
+    if (*tokenEnd == '%') { // pattern
+      ++tokenEnd;
+      if (tokenEnd == end) {
+        break;
+      }
+      switch (*tokenEnd) {
+        case 'a':
```

Comment:
It is extra work and not sue if that will make it better, but I was thinking if it worth getting them into "constexpr char" and using constants?

Reply:
I think it's worth it.

Reply:
After rethinking about it I think it is actually better to just leave the literal letters in the switch statement. They are specifiers without names. There is probably no need to make them constexpr char. Nowhere is going to reuse them either.

Reply:
My idea was not about a reuse, but more like about code organization, but I agree that it probably not worth it.